src/amo/components/SearchForm.js

```diff
@@ -23,12 +26,31 @@ export class SearchFormBase extends React.Component {
     i18n: PropTypes.object.isRequired,
     pathname: PropTypes.string.isRequired,
     query: PropTypes.string.isRequired,
+    suggestions: PropTypes.arrayOf(PropTypes.string).isRequired,
+    dispatch: PropTypes.func.isRequired,
+    errorHandler: PropTypes.object.isRequired,
   }
 
   static contextTypes = {
     router: PropTypes.object,
   }
 
+  constructor(props: Object) {
+    super(props);
+
+    this.state = {
```

Comment:
We tend not to use internal component state for anything. I know we had an issue open about removing `setState` usage from existing components (#1250) which I'm not sure is finished, but I don't think we should add any new `setState` usage. I'd rather everything be in the redux store instead.